tests/framework/prototype.py

```diff
@@ -586,10 +587,15 @@ def setUp(self):
     def runtest(self):
         """Run a configuration test"""
         self.cr_event.trigger()
+        self.passed = True
 
     def tearDown(self) -> None:
-        try:
-            self.cr_event.clean_up()
-        except Exception as ex:
-            logger.error(str(ex))
-            K8S_CLUSTER_MANAGER.cleanup()
+        if self.passed:
+            try:
+                self.cr_event.clean_up()
+            except Exception as ex:
+                logger.error(str(ex))
+                K8S_CLUSTER_MANAGER.cleanup()
+        else:
+            namespace = self.cr_event.namespace
+            logger.error("Preserving CR in namespace %s for debugging", namespace)
```

Comment:
nit: 
- Could we perhaps consider showing users how to clean up, as they will need to manually handle these tasks? 
- we can directly use `self.cr_event.namespace`.
- I feel we should use `logger.info` instead of `logger.error`. Error information is already captured in previous failed test.